[feat]: [DOC-3356]: Adding documentation for the strategy expressions to point to specific strategy levels (#2343)
harness/developer-hub: docs/platform/12_Variables-and-Expressions/harness-variables.md

This expression returns the `identifierPostFix` of the step group when the current node is under the step group, or when the current node is the step group itself, and that step group is a child of a looping strategy.

> ### <+strategy.node.STRATEGY_NODE_IDENTIFIER.identifierPostFix>

Reviewer: @brijeshdhakar what is identifierpostfix?

Author: Its defined in this class above the current change. 
### identifierPostFix overview, Line:1358

> This expressions retrieves the `identifierPostFix` for the node that is child of looping strategy with identifier `STRATEGY_NODE_IDENTIFIER`

Reviewer: ```suggestion
This expression retrieves the `identifierPostFix` for the node that is the child of a looping strategy with the identifier `STRATEGY_NODE_IDENTIFIER`.
```

> For example, let's consider two nested step groups, sg1 and sg2 (child of sg1). Both sg1 and sg2 have a looping strategy configured. Using the expression, `<+stepGroup.identifierPostFix>` always retrieves the `identifierPostFix` of sg2. 

Reviewer: ```suggestion
For example, let's consider two nested step groups, sg1 and sg2 (child of sg1). Both sg1 and sg2 have a looping strategy configured. The expression, `<+stepGroup.identifierPostFix>` always retrieves the `identifierPostFix` of sg2. 
```

> To obtain the `identifierPostFix` for a specific step group, you can use the following expressions:

Reviewer: ```suggestion
Use the following expressions to obtain the `identifierPostFix` for a specific step group:
```

> For example, let's consider a scenario with two nested StepGroups: sg1 and sg2 (child of sg1). Both sg1 and sg2 have a looping strategy configured. In this case, using the expression `<+stepGroup.identifierPostFix>` would always retrieve the `identifierPostFix` of sg2. To obtain the `identifierPostFix` for specific StepGroup, you can use the following expressions:

Reviewer 2: Can we add a diagram or screenshot here please? Users will want to refer back to it as they work through the examples below.

> For example, let's consider a scenario with two nested StepGroups: sg1 and sg2 (child of sg1). Both sg1 and sg2 have a looping strategy configured. In this case, using the expression `<+stepGroup.identifierPostFix>` would always retrieve the `identifierPostFix` of sg2. To obtain the `identifierPostFix` for specific StepGroup, you can use the following expressions:

Reviewer: ```suggestion
For example, let's consider two nested step groups, sg1 and sg2 (child of sg1). Both sg1 and sg2 have a looping strategy configured. Using the expression, `<+stepGroup.identifierPostFix>` always retrieves the `identifierPostFix` of sg2. To obtain the `identifierPostFix` for a specific step group, you can use the following expressions:
```

> ```

Reviewer: ```suggestion

```

> <+strategy.node.sg1.identifierPostFix>: Retrieves the identifierPostFix for the node with the identifier sg1(Parent stepGroup).

Reviewer: ```suggestion
`<+strategy.node.sg1.identifierPostFix>`: Retrieves the `identifierPostFix` for the node with the identifier sg1 (parent step group).
```

> <+strategy.node.sg2.identifierPostFix>: Retrieves the identifierPostFix for the node with the identifier sg2(Child stepGroup).

Reviewer: ```suggestion
`<+strategy.node.sg2.identifierPostFix>`: Retrieves the `identifierPostFix` for the node with the identifier sg2 (child step group).
```

> ```

Reviewer: ```suggestion

```

Similarly, you can use other strategy expressions for any specific strategy level if a looping strategy is configured for both the parent and child nodes.

> ### <+strategy.node.STRATEGY_NODE_IDENTIFIER.*>

Reviewer: ```suggestion
### <+strategy.node.STRATEGY_NODE_IDENTIFIER.*>

```

> By following the provided format, you can retrieve the values of any strategy expressions associated with looping strategies at various levels. This is especially useful when looping strategies are configured within nested levels. Let's take a look at some examples:

Reviewer: ```suggestion
Using this format, you can retrieve the values of any strategy expressions associated with looping strategies at various levels. This is useful when looping strategies are configured within nested levels. 

Here are some examples:
```

> ```

Reviewer: ```suggestion

```

> <+strategy.node.sg1.iteration>: Retrieves the current iteration for the node with the identifier sg1(Parent stepGroup).

Reviewer: ```suggestion
* `<+strategy.node.sg1.iteration>`: Retrieves the current iteration of the node with the identifier sg1 (parent step group).
```

> <+strategy.node.sg2.iteration>: Retrieves the current iteration for the node with the identifier sg2(Child stepGroup).

Reviewer: ```suggestion
* `<+strategy.node.sg2.iteration>`: Retrieves the current iteration of the node with the identifier sg2 (child step group).
```

<+strategy.node.some_node_with_looping_strategy.iteration>: Retrieves the current the iteration for the node with identifier some_node_with_looping_strategy.(some_node_with_looping_strategy can be any type of node stage/step/stepGroup)

> <+strategy.node.sg1.iterations>: Retrieves the total iterations for the node with the identifier sg1(Parent stepGroup).

Reviewer 2: What do we mean by "total iterations"? Is it an array of all the iterations, or is it some kind of summary?
